src/core/sig_state.ml

```diff
@@ -24,11 +24,39 @@ type sig_state =
 
 type t = sig_state
 
+(** Symbols and signature used for unification rules. *)
+module Unif_rule =
+  struct
+
+    (** Ghost signature holding the symbols used in unification rules. This
+        signature is an automatic dependency of all other signatures, and is
+        automatically loaded. *)
+    let sign : Sign.t =
+      let open Files in
+      let pth = Path.ghost "unif_rule" in
+      let s = Sign.create pth in
+      (* Remove the dependency on itself. *)
+      s.sign_deps := PathMap.remove pth !(s.sign_deps);
+      Sign.loaded := Files.PathMap.add pth s !(Sign.loaded);
+      s
```

Comment:
To have \#equiv and \#list printed as "≡" and "," respectively, it would be better to declare them as binops too, with "," associative. Btw, it would then be better to rename \#list into \#comma. Then, there would be no need to have a special parser rule. Finally, you should call open_sign at some point.

Reply:
Ah but, currently, "," cannot be used in an identifier... So, indeed, we need a special parser rule.

Reply:
OK, so it would rather be a cons symbol than a list symbol.

Reply:
Right. Otherwise you won't have a nice printing (unless you use a specific printing function).

Reply:
I don't think we need to call `open_sign` since `of_sign` initialises "correctly" the fields.

Reply:
No, in_scope and pp_hints are not set correctly.

Reply:
OK, it should be now that I `open_sign` when a `sig_state` is created.

Reply:
Where do you do this? I cannot find it.

Reply:
in `of_sign`